File: ChatMicroservice/ChatMicroservice/src/main/java/com/example/ChatMicroservice/entity/ChatMessage.java
package com.example.ChatMicroservice.entity;

import lombok.*;

import java.util.Date;



@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ChatMessage {
    private String chatId;
    private String senderName;
    private String receiverName;
    private String message;
    private Date timestamp;
    private Status status;
}

File: frontend/src/app/models/ChatMessage.ts
import {Status} from './Status';

export interface ChatMessage {
  chatId: string; // Unique identifier for the chat
  senderName: string; // Name of the sender
  receiverName: string; // Name of the receiver
  message: string; // The chat message content
  timestamp: Date; // The date and time of the message
  status: Status; // The status of the message
}
